src/averageAges.js

```diff
@@ -55,7 +72,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let children = people.filter(
+    child => people.some(mother => mother.name === child.mother));
+  const childIsSon = getSex(children, 'm');
+
+  children = !onlyWithSon ? children : childIsSon;
+
+  const averageAgeDiff = children.reduce(
+    (sum, kid) => sum + (kid.born - people.find(
+      mom => kid.mother === mom.name).born), 0) / children.length;
```

Comment:
I'd rather decompose it this way
```suggestion
  const ageDifferences = reduce((sum, kid) => {
    const mother = people.find(({ name })) => kid.mother === name);
    const ageDifference = kid.born - mother.born;
    
    return sum + ageDifference;
  }, 0)
```